src/Compiler/RefC/CC.idr

```diff
@@ -0,0 +1,74 @@
+module Compiler.RefC.CC
+
+import Core.Context
+import Core.Options
+
+import System
+
+import Idris.Version
+import Libraries.Utils.Path
+
+public export
+findCC : IO String
+findCC
+    = do Nothing <- getEnv "IDRIS2_CC"
+           | Just cc => pure cc
+         Nothing <- getEnv "CC"
+           | Just cc => pure cc
+         pure "cc"
+
+fullprefix_dir : Dirs -> String -> String
+fullprefix_dir dirs sub
+    = prefix_dir dirs </> "idris2-" ++ showVersion False version </> sub
+
+export
+compileCObjectFile : {auto c : Ref Ctxt Defs}
+                  -> {default False asLibrary : Bool}
+                  -> (sourceFile : String)
+                  -> (objectFile : String)
+                  -> Core (Maybe String)
+compileCObjectFile {asLibrary} sourceFile objectFile =
+  do cc <- coreLift findCC
+     dirs <- getDirs
+
+     let libraryFlag = if asLibrary then "-fpic " else ""
+
+     let runccobj = cc ++ " -c " ++ libraryFlag ++ sourceFile ++
+                       " -o " ++ objectFile ++ " " ++
+                       "-I" ++ fullprefix_dir dirs "refc " ++
+                       "-I" ++ fullprefix_dir dirs "include"
+
+     coreLift $ putStrLn runccobj
```

Comment:
Should this `putStrLn` be a call to `log "compile.refc"` instead?
(You'll need to import `Core.Context.Log` to get access to it)

Reply:
This line is unchanged from the original code, just moved. I'm happy to change it if that's the right thing to do, but I don't know why this line is as it is.

Reply:
Right but as we are working on that part of the codebase, that's an opportunity
to clean it up a bit as we go. Debugging information should be communicated using
`log` rather than `putStrLn`. This way it can be turned off if we don't need the
info.